src/lightning/pytorch/trainer/connectors/logger_connector/result.py

```diff
@@ -357,6 +357,24 @@ def log(
         rank_zero_only: bool = False,
     ) -> None:
         """See :meth:`~lightning.pytorch.core.module.LightningModule.log`"""
+        # if value is a Metric and sync_dist != None, then
+        if isinstance(value, Metric) and sync_dist is not None:
```

Comment:
I wouldn't run this logic if `sync_dist = False`. In that case let the metric be run with the configuration it has already

Reply:
Makes sense, change it.